pythia/seo/ELO.py

```diff
@@ -0,0 +1,195 @@
+            `0` denotes less complex, `1` denotes more complex
+
+        Notes
+        -----
+        To make updates in the original rankings DataFrame, for each classification,
+        two state dictionaries need to be maintained, corresponfing to the two AR images.
+        The changes are made to these state dictionaries and then the ranking DataFrame is updated.
+        """
+        # state dicts
+        state_dict_0 = self.rankings.loc[image_0].to_dict()
+        state_dict_0['last_scores'] = list(map(float, state_dict_0['last_scores'].split(',')))
+        state_dict_1 = self.rankings.loc[image_1].to_dict()
+        state_dict_1['last_scores'] = list(map(float, state_dict_1['last_scores'].split(',')))
+
+        expected_score_0 = self.expected_score(image_0, image_1)
+        expected_score_1 = 1 - expected_score_0
+
+        new_rating_0 = self.new_rating(image_0, score_for_image_0, expected_score_0)
+        new_rating_1 = self.new_rating(image_1, 1 - score_for_image_0, expected_score_1)
+
+        state_dict_0['last_scores'].append(new_rating_0)
+        state_dict_1['last_scores'].append(new_rating_1)
+
+        if len(state_dict_0['last_scores']) > self.score_memory:
+            state_dict_0['last_scores'].pop(0)
+        if len(state_dict_1['last_scores']) > self.score_memory:
+            state_dict_1['last_scores'].pop(0)
+
+        new_std_dev_0 = min(np.std(state_dict_0['last_scores']), 1000000)  # prevents Infinity
+        new_k_0 = min(max(new_std_dev_0, self.score_change['min']), self.score_change['max'])
+
+        new_std_dev_1 = min(np.std(state_dict_1['last_scores']), 1000000)  # prevents Infinity
+        new_k_1 = min(max(new_std_dev_1, self.score_change['min']), self.score_change['max'])
+
+        # Updating Data
+        state_dict_0['image_id'] = [state_dict_0['image_id']]  # necessary to make the Dataframe index
+        state_dict_0['score'] = new_rating_0
+        state_dict_0['std_dev'] = new_std_dev_0
+        state_dict_0['k_value'] = new_k_0
+        state_dict_0['count'] += 1
+        state_dict_0['last_scores'] = ",".join(map(str,state_dict_0['last_scores']))  # Storing the list of states as a String
+
+        state_dict_1['image_id'] = [state_dict_1['image_id']]  # necessary to make the Dataframe index
+        state_dict_1['score'] = new_rating_1
+        state_dict_1['std_dev'] = new_std_dev_1
+        state_dict_1['k_value'] = new_k_1
+        state_dict_1['count'] += 1
+        state_dict_1['last_scores'] = ",".join(map(str,state_dict_1['last_scores']))  # Storing the list of states as a String
+
+        # Making the Update DataFrames
+        update_df_0 = pd.DataFrame(state_dict_0, index=state_dict_0['image_id'])
+        update_df_1 = pd.DataFrame(state_dict_1, index=state_dict_0['image_id'])
+
+        # Updating the original DataFrame
+        self.rankings.update(update_df_0)
+        self.rankings.update(update_df_1)
```

Comment:
This function repeats everything twice, how can we make it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)er?